wallet/src/account/output_cache/mod.rs

```diff
@@ -680,56 +684,86 @@ impl OutputCache {
+                    match cmd {
+                        AccountCommand::MintTokens(_, _)
+                        | AccountCommand::UnmintTokens(_)
+                        | AccountCommand::LockTokenSupply(_)
+                        | AccountCommand::ChangeTokenMetadataUri(_, _)
+                        | AccountCommand::ChangeTokenAuthority(_, _)
+                        | AccountCommand::UnfreezeToken(_)
+                        | AccountCommand::ConcludeOrder(_)
+                        | AccountCommand::FillOrder(_, _, _) => { /* do nothing */ }
+                        | AccountCommand::FreezeToken(token_id, _) => {
+                            frozen_token_id = Some(*token_id);
+                        }
+                    }
+                }
+            }
         }
 
-        let frozen_token_id = tx.inputs().iter().find_map(|inp| match inp {
-            TxInput::Utxo(_) | TxInput::Account(_) => None,
-            TxInput::AccountCommand(_, cmd) => match cmd {
-                AccountCommand::MintTokens(_, _)
-                | AccountCommand::UnmintTokens(_)
-                | AccountCommand::LockTokenSupply(_)
-                | AccountCommand::ChangeTokenMetadataUri(_, _)
-                | AccountCommand::ChangeTokenAuthority(_, _)
-                | AccountCommand::UnfreezeToken(_)
-                | AccountCommand::ConcludeOrder(_)
-                | AccountCommand::FillOrder(_, _, _) => None,
-                AccountCommand::FreezeToken(frozen_token_id, _) => Some(frozen_token_id),
-            },
-        });
-
+        // Collect all conflicting txs
         let mut conflicting_txs = vec![];
-        if let Some(frozen_token_id) = frozen_token_id {
+
+        if frozen_token_id.is_some() || confirmed_account_nonce.is_some() {
             for unconfirmed in self.unconfirmed_descendants.keys() {
-                let unconfirmed_tx = self.txs.get(unconfirmed).expect("must be present");
-                if self.uses_token(unconfirmed_tx, frozen_token_id) {
-                    let unconfirmed_tx = self.txs.get_mut(unconfirmed).expect("must be present");
-                    match unconfirmed_tx {
-                        WalletTx::Tx(ref mut tx) => {
-                            tx.set_state(TxState::Conflicted(block_id));
+                if let Some(frozen_token_id) = frozen_token_id {
+                    let unconfirmed_tx = self.txs.get(unconfirmed).expect("must be present");
+                    if self.uses_token(unconfirmed_tx, &frozen_token_id) {
+                        if let WalletTx::Tx(tx) = unconfirmed_tx {
+                            conflicting_txs.push(tx.get_transaction().get_id());
                         }
-                        WalletTx::Block(_) => {}
-                    };
+                    }
+                }
 
-                    conflicting_txs.push(unconfirmed);
+                if let Some((confirmed_account, confirmed_account_nonce)) = confirmed_account_nonce
```

Comment:
this can probably be an else if, as the transactions that use a token id will be a superset of the transactions that use the token id and have a specific nonce.

Reply:
I think that it'd make the code more confusing especially when modifying in the future. I replaced the vec with set to avoid duplicates.